inventory_management_system_api/auth/jwt_bearer.py

```diff
@@ -64,3 +64,18 @@ def _is_jwt_access_token_valid(self, access_token: str) -> bool:
             payload = None
 
         return payload is not None and "username" in payload
+
+    def is_jwt_access_token_authorised(self, access_token: str) -> bool:
+        """
+        Check if the JWT access token is authorised.
+
+        It does this by checking that the token's payload contains roles, and that (at least) one of
+        these roles is one of the configured privileged_roles
+
+        :param access_token: The JWT access token to check
+        :return `True` if the JWT access token's payload contains roles, and overlaps the configured
+        privileged_roles, `False` otherwise.
+        """
+        logging.info("Checking if JWT access token is authorised for operation")
+        payload = jwt.decode(access_token, options={"verify_signature": False})
```

Comment:
What will happen if the decode here fails? How does the endpoint behave?

Reply:
See my comment https://github.com/ral-facilities/inventory-management-system-api/pull/602#discussion_r2406192598